metaflow/runtime.py

```diff
@@ -749,32 +775,44 @@ def __init__(
         self.datastore_sysroot = flow_datastore.datastore_root
         self._results_ds = None
 
+        # Only used in clone-only resume.
+        self._is_resume_leader = None
+        self._resume_done = None
+        self._resume_identifier = resume_identifier
+
         origin = None
         if clone_run_id and may_clone:
             origin = self._find_origin_task(clone_run_id, join_type)
         if origin and origin["_task_ok"]:
             # At this point, we know we are going to clone
             self._is_cloned = True
+
+            if resume_identifier:
```

Comment:
we can probably move this into reentrant case. It should be set if we are reentrant. This will also make it slightly cleaner insofar as reentrant is the main flag controlling this distributed resume thingy.

Reply:
moved under reentrant, and raise error if not set.